vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimSearchGroupBase.kt

```diff
@@ -48,23 +48,35 @@ public abstract class VimSearchGroupBase : VimSearchGroup {
     /**
      * Last string trailing a pattern. E.g. in '/pattern/e+2', 'e+2' is trailing.
      */
-    private var lastPatternTrailing: String? = ""
+    public var lastPatternTrailing: String? = ""
 
     /**
      * Last used search direction.
      */
-    private var lastDirection: Direction = Direction.FORWARDS
+    @JvmStatic
+    protected var lastDirection: Direction = Direction.FORWARDS
+      @JvmStatic
+      get
+      @JvmStatic
+      set
 
     /**
      * The type of the last used pattern.
      */
-    private var lastPatternType: PatternType? = null
+    @JvmStatic
```

Comment:
Why do we need all new `@JvmStatic`? I see that their usage was moved from java class to kotlin class.